edgelet/edgelet-http-mgmt/src/server/module/prepare_update.rs

```diff
@@ -49,13 +50,33 @@ where
             })
             .and_then(|(core_spec, runtime)| {
                 let name = core_spec.name().to_string();
-                runtime.registry().pull(core_spec.config()).then(|result| {
-                    result.with_context(|_| ErrorKind::PrepareUpdateModule(name.clone()))?;
-                    Ok(name)
-                })
+                let pull_policy = *core_spec.pull_policy();
```

Comment:
You don't need the `*` since match ergonomics will automatically change the match arm patterns to refs.

Reply:
So, I do need to use the deref operator here cause if I remove it, the compiler complains about core_spec going out of scope while still being borrowed at the end of the and_then() block.
Additionally, Clippy suggests using the deref operator instead of calling clone() on core_spec.pull_policy() as the type is Copy(able)

Reply:
>So, I do need to use the deref operator here cause if I remove it, the compiler complains about core_spec going out of scope while still being borrowed at the end of the and_then() block.

Yes, because you also need to deref `pull_policy` when you put it in the tuple and return it. Otherwise you're returning `(_, &PullPolicy)`

>Additionally, Clippy suggests using the deref operator instead of calling clone() on core_spec.pull_policy() as the type is Copy(able)

Of course, but I also never said anything about calling `.clone()`

Reply:
Discussed offline. It's fine as it is.